sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ContainerConnectionConfig.java

```diff
@@ -0,0 +1,28 @@
+package com.azure.cosmos;
+
+import java.util.Collections;
+import java.util.Set;
+
+public final class ContainerConnectionConfig {
+    private final String containerLink;
```

Comment:
I don't think these config settings make much sense - partitioning isn't even always done per container - for shared database connections partitioning is done on database level. Also exposing MaxChannelsPerReplica is leaking implementation details into public surface area. Please let's discuss these changes before spending any more time on this PR.